### PR TITLE
Add test for CVE-2019-14287

### PR DESCRIPTION
**Pull request by pvdrz**
This PR adds a test for CVE-2019-14287.

### CODE REVIEW

File: test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs [hunk 1]
```diff
@@ -451,3 +451,33 @@ fn supplemental_group_matching() -> Result<()> {
...
+
+    let env = Env(format!(
+        "{USERNAME} {host} = (ALL, !root) NOPASSWD: /usr/bin/id"
+    ))
+    .user(User(USERNAME))
+    .group(GROUPNAME)
+    .hostname(host)
+    .build()?;
+
+    for uid in [(-1).to_string(), u32::MAX.to_string()] {
```
Comment (japaric) [inline test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs:469]: nit: you could refactor the `to_string` into the for loop if you write the array as `[-1, u32::MAX as i64]`

File: test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs [hunk 2]
```diff
@@ -451,3 +451,33 @@ fn supplemental_group_matching() -> Result<()> {
...
+        let output = Command::new("sudo")
+            .args(["-u", &format!("#{uid}"), "id", "-u"])
+            .output(&env)?;
+
+        assert!(!output.status().success());
+        if sudo_test::is_original_sudo() {
+            assert_contains!(output.stderr(), "unknown user");
```
Comment (japaric) [inline test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs:476]: nit: you could refactor the `assert_contains` if you have the `if` expression return the "needle" string into a variable

File: test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs [hunk 3]
```diff
@@ -451,3 +451,33 @@ fn supplemental_group_matching() -> Result<()> {
...
+/// This test tracks [CVE-2019-14287](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-14287)
+/// which is explained in more detail [here](https://www.sudo.ws/security/advisories/minus_1_uid/).
+#[test]
+fn minus_1_uid() -> Result<()> {
+    let host = "myhost";
+
+    let env = Env(format!(
+        "{USERNAME} {host} = (ALL, !root) NOPASSWD: /usr/bin/id"
+    ))
+    .user(User(USERNAME))
```
Comment (japaric) [inline test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs:464]: nit: this should be `rustfmt`-ed

**Review by japaric — APPROVED** [review]
LGTM. left some nit comments

File: test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs (hunk 3 above)
Comment (pvdrz) [reply test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs:464]: That's how my rustfmt formats it :sweat_smile:

**Pull request merged**